### PR TITLE
Implement glossary functionality

### PR DESCRIPTION
**Pull request by shravanhariharan2**
Implementing glossary functionality by pulling all trees in alphabetic order w/ database index on names to keep efficiency.

Migrations for db tables will be added as well

### CODE REVIEW

**Review by trulyronak — APPROVED** [review]
could improve on adding status codes for errors, give more info when you fail

status codes could reflect what failed

functionality lgtm

File: app/api/treeRouter.py [hunk 1]
```diff
@@ -3,44 +3,67 @@
...
-    body = request.get_json()
-    if not body:
-        return "Error: No request body provided"
-    nodes = treeService.addPointsById(body["id"], body["points"])
-    return jsonify(nodes)
+    try:
+        body = request.get_json()
+        if not body:
+            return sendError(400, "No request body provided")
+        tree = treeService.addPointsById(body["id"], body["points"])
```
Comment (sumeet-bansal) [inline app/api/treeRouter.py]: Check for `body['id']` and `body['points']`.

File: app/config.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class Config:
     SECRET_KEY = os.environ["SECRET_KEY"]
     SQLALCHEMY_TRACK_MODIFICATIONS = False
     SQLALCHEMY_DATABASE_URI = postgres_uri
+    MAX_CONTENT_LENGTH = 512 * 1024
```
Comment (sumeet-bansal) [inline app/config.py]: Specify units: `MAX_CONTENT_SIZE_IN_BYTES`.

**Review by sumeet-bansal — APPROVED** [review]

File: app/config.py (hunk 2 above)
Comment (shravanhariharan2) [reply app/config.py]: `MAX_CONTENT_LENGTH` is a field that `werkzeug` expects in the app config for the `RequestEntityTooLarge` exception to be properly thrown, so I just added a conversion constant to make the conversion clearer

File: app/api/treeRouter.py [hunk 3]
```diff
@@ -44,8 +44,8 @@ def putTree(tree_id):
 def addPoints():
     try:
         body = request.get_json()
-        if not body:
-            return sendError(400, "No request body provided")
+        if not body or body["id"] or body["points"]:
```
Comment (sumeet-bansal) [inline app/api/treeRouter.py]: I'd expect the `not` to only apply to `body` so you'd end up returning an error if `body['id']` or `body['points']` is present.
[
Screen Shot 2020-05-27 at 4 33 52 PM
](url)

**Review by sumeet-bansal — CHANGES_REQUESTED** [review]

File: app/api/treeRouter.py (hunk 3 above)
Comment (shravanhariharan2) [reply app/api/treeRouter.py]: whoops thanks, turns out I was also getting a `KeyError` for accessing "id" or "points" if it wasn't passed in too so fixed that as well.

**Review by sumeet-bansal — APPROVED** [review]

**Pull request merged**